### PR TITLE
Fix url for proxies on Windows

### LINKED ISSUE #2796: path.join in proxy blueprint fails on windows
In the proxy blueprint we have

    req.url = path.join(proxyPath, req.url);

But this would seem to be incorrect. `path.join` does a normalization which under Windows changes slashes to backslashes (ouch!). `path.join` should not be used for URLs, only for file system paths.

This was introduced as part of commit https://github.com/stefanpenner/ember-cli/commit/431e6986d583f2a18522c67529058495d246259f made by @tstirrat.

### PR DESCRIPTION
**Pull request by chadhietala**
Should close #2796

### CODE REVIEW

**Comment by chadhietala** [comment]
@stefanpenner :+1: or :-1: ?

File: blueprints/http-proxy/files/server/proxies/__name__.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = function(app) {
 
   app.use(proxyPath, function(req, res, next){
     // include root path in proxied request
-    req.url = path.join(proxyPath, req.url);
+    req.url = urlJoin(proxyPath, req.url);
```
Comment (stefanpenner) [inline blueprints/http-proxy/files/server/proxies/__name__.js]: im curious why we can just `proxyPath +  '/' + req.url`

Comment (chadhietala) [inline blueprints/http-proxy/files/server/proxies/__name__.js]: Nope just over doing it to the nth degree.

**Comment by chadhietala** [comment]
Not really sure what is going on with CI.

**Comment by stefanpenner** [comment]
many CI things are fixed, can you rebase from master and lets see what happens.

**Comment by stefanpenner** [comment]
pulled in another simpler PR that does this.

**Pull request closed without merging**